tools/d2g/d2g.go

```diff
@@ -84,7 +87,7 @@ func ConvertTaskDefinition(dwTaskDef json.RawMessage) (json.RawMessage, error) {
 // equivalent Generic Worker scopes. These scopes should be used together with
 // a converted Docker Worker task payload (see d2g.Convert function) to run
 // Docker Worker tasks under Generic Worker.
-func Scopes(dwScopes []string) (gwScopes []string) {
+func Scopes(dwScopes []string, dwTaskDef json.RawMessage) (gwScopes []string, err error) {
```

Comment:
I think we should pass in a DockerWorkerPayload struct here, rather than json.RawMessage. The only reason we do that elsewhere (ConvertTaskDefinition) is because the task definition can evolve independently of the gw/dw payload definitions (e.g. if new properties are added, that the queue cares about, but the worker doesn't). However, a version of d2g only supports specific docker-worker payload format, and can only generate properties in the generic worker payload format that it knows about, so with payloads, it would be better to pass around the go types, for consistency with the Convert method, etc.

Reply:
This is the entire task definition, though. I agree, if I only needed the payload, that I should pass in the go type.